dev/TabView/TabViewItem.cpp

```diff
@@ -106,15 +129,90 @@ winrt::AutomationPeer TabViewItem::OnCreateAutomationPeer()
     return winrt::make<TabViewItemAutomationPeer>(*this);
 }
 
+void TabViewItem::OnCloseButtonOverlayModeChanged(winrt::TabViewCloseButtonOverlayMode const& mode)
+{
+    m_closeButtonOverlayMode = mode;
+    if (mode == winrt::TabViewCloseButtonOverlayMode::OnHover && !IsSelected())
+    {
+        UpdateCloseButtonVisibility(winrt::Visibility::Collapsed);
+    }
+    if (mode == winrt::TabViewCloseButtonOverlayMode::Persistent || mode == winrt::TabViewCloseButtonOverlayMode::Auto)
+    {
+        UpdateCloseButtonVisibility(winrt::Visibility::Visible);
+    }
+}
+
+void TabViewItem::OnTabViewWidthModeChanged(winrt::TabViewWidthMode const& mode)
+{
+    m_tabViewWidthode = mode;
+
+    // When switching to compact, all items hide the labels and only show their icon
+    // This switch needs to be done here!
+    if (mode == winrt::TabViewWidthMode::Compact)
+    {
+        if (!IsSelected())
+        {
+            winrt::VisualStateManager::GoToState(*this, L"Compact", false);
+        }
+    }
+    else
+    {
+        winrt::VisualStateManager::GoToState(*this, L"StandardWidth",false);
+    }
+}
+
 
 void TabViewItem::UpdateCloseButton()
 {
+    if (!IsClosable())
+    {
+        UpdateCloseButtonVisibility(winrt::Visibility::Collapsed);
+    }
+    else
+    {
+        switch (m_closeButtonOverlayMode)
+        {
+            case winrt::TabViewCloseButtonOverlayMode::Persistent:
+            {
+                // In case of "Persistent" always show button
+                UpdateCloseButtonVisibility(winrt::Visibility::Visible);
+                break;
+            }
+            case winrt::TabViewCloseButtonOverlayMode::Auto:
+            {
+                // In case of "Auto" always show button
+                UpdateCloseButtonVisibility(winrt::Visibility::Visible);
+                break;
+            }
```

Comment:
These are the same and should be combined.

Reply:
My idea was that if we want to change the "Auto" behavior, we already have the code lined up for that change. But sure, I will refactor that.